src/java/org/apache/cassandra/index/sai/disk/PerIndexWriter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk;
+
+import java.io.IOException;
+
+import com.google.common.base.Stopwatch;
+
+import org.apache.cassandra.db.rows.Row;
+import org.apache.cassandra.index.sai.utils.PrimaryKey;
+
+/**
+ * Creates an on-disk index for a given index.
+ */
+public interface PerIndexWriter
```

Comment:
nit: In both class naming and JavaDoc here, some indication that this is column-level would be helpful.

Reply:
> My concern is that if we make this change to the class name do we then change SSTableIndex -> SSTableColumnIndex.

Fine w/ me 🤷 (I may have even already suggested this...so many comments, can't remember)

> My preference would be to make this existing terminology clear in a top level README file.

At the end of the day, I'm fine with either or both.